fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SubspaceProvider.java

```diff
@@ -23,28 +23,44 @@
 import com.apple.foundationdb.annotation.API;
 import com.apple.foundationdb.record.logging.LogMessageKeys;
 import com.apple.foundationdb.subspace.Subspace;
-
 import javax.annotation.Nonnull;
 import java.util.concurrent.CompletableFuture;
 
 /**
  * Subspace provider can provide a subspace (might be blocking) and logging information to the subspace (non-blocking).
  */
-@API(API.Status.MAINTAINED)
+@API(API.Status.INTERNAL)
```

Comment:
This looks good, but the implementations should probably also mark themselves as `INTERNAL`.